.github/workflows/ci.yml

```diff
@@ -57,65 +60,69 @@ jobs:
         run: |
           cd Consul.Test
           SYSTEM=$(uname | sed 's/MINGW.*/windows/' | tr A-Z a-z)
-          curl -sSL https://releases.hashicorp.com/consul/${CONSUL_VERSION}/consul_${CONSUL_VERSION}_${SYSTEM}_amd64.zip -o consul.zip
+          curl -sSL https://releases.hashicorp.com/consul/${{ matrix.consul }}/consul_${{ matrix.consul }}_${SYSTEM}_amd64.zip -o consul.zip
           unzip consul.zip
           rm consul.zip
       - name: Build
-        run: dotnet build --configuration=Release -p:TreatWarningsAsErrors=true
+        run: dotnet build Consul.Test --configuration=Release -p:TreatWarningsAsErrors=true --framework=${{ matrix.framework }}
       - name: Run tests
         shell: bash
         run: |
-          cd Consul.Test
-          ./consul agent -dev -config-file test_config.json -log-file consul.log >consul-stdout.log 2>consul-stderr.log &
-          dotnet test --configuration=Release --logger "GitHubActions;report-warnings=false" --no-build -v=Normal
+          ./Consul.Test/consul agent -dev -config-file Consul.Test/test_config.json -log-file consul.log >consul-stdout.log 2>consul-stderr.log &
+          dotnet test Consul.Test --configuration=Release --logger "GitHubActions;report-warnings=false" --no-build -v=Normal --framework=${{ matrix.framework }}
       - name: Upload Consul logs
         if: failure()
         uses: actions/upload-artifact@v2
         with:
-          name: consul-logs-${{ matrix.os }}
-          path: Consul.Test/consul*.log
-      - name: Create NuGet package
-        if: runner.os == 'Windows'
-        run: |
-          if (-not ("${{ github.ref }}" -like "refs/tags/v*")) {
-            # WORKAROUND: Add letter prefix to ensure that MSBuild treats
-            # the suffix as a string. e.g. '0313071' will fail as an invalid
-            # version string, but 'G0313071' will succeeded. It looks like
-            # the parser does not like numbers with a leading zero.
-            $suffix = 'g' + $(git rev-parse --short HEAD)
-            $params = "--version-suffix", $suffix
-          }
+          name: consul-logs-${{ matrix.consul }}-${{ matrix.framework }}-${{ matrix.os }}
+          path: consul*.log
 
-          dotnet pack --configuration=Release --no-build @params Consul/Consul.csproj
+  package:
```

Comment:
`package` and `publish` won't work once we want to release Consul.AspNetCore, which is a regression from the previous CI. I know we can't yet publish Consul.AspNetCore (I need to talk to the current package owner), but we should keep our CI ready for it.

Reply:
Thanks to @zlzforever, we now have ownership of Consul.AspNetCore on nuget.org, so this really needs to work now 😄

Reply:
Not really a regression technically, as there was no continuous delivery for the asp integration :-P

Anyway, I sort of fixed that now. But we're going to need to decide on the deploy process here. If one tag and shared version number for both packages, it is easy. If packages have different package versions, different automated releases, and/or different tags, well... some decisions need to be made :-)

Reply:
There is/was a CD for it, we just never pulled the trigger on making a release 😔
https://github.com/G-Research/consuldotnet/blob/2e7b1b15c8bac03d1ff1b8f47ef578567ffea3f3/.github/workflows/consul.aspnetcore.yml#L62-L76

The decision back then was to use a different tag for releasing the ASP library (`Consul.AspNetCore_vX.Y.Z`), but I'm more than open to revisit this.

The way is see it at the moment is that using the same version numbers would probably mean having many releases of the ASP library that have no actual changes. On the other hand, having different version numbers means using those ugly tags.

If we do go for different version numbers, I think this 'initial' release should be 2.0.0, as the existing 1.0.0 package was a different project altogether.

@marcin-krystianc @mfkl thoughts?

Reply:
> The way is see it at the moment is that using the same version numbers would probably mean having many releases of the ASP library that have no actual changes.

True, but that also means they are kept in sync by design, and provides a straightforward versioning scheme to end user.

My opinion is to have them share the version numbers, if I recall correctly Marcin prefers the different version number approach. Anyway, it is not highly critical, I guess :-)

Reply:
Both approaches have pros & cons. I think we are at the point were we can try using the same version for both libraries (1.x) and if we decide later that we don't like it we can change the version of AspNetCore to v2.x. But the other order is not possible, once we publish version 2.x we will not be able to switch back to v1.x...
Thus maybe we should try using the same version for both libraries and see how it goes ?

Reply:
Sounds sensible to me, let's go for same version!

Reply:
Ok